topics/wallet-iteration-standard/ledger-extension.md

```diff
@@ -0,0 +1,89 @@
+# ICRC-27: Ledger extension for Wallet Interaction Standard
+
+| Status |
+|:------:|
+| Draft  |
+
+## Summary
+
+This specification describes a ledger extension for the ICRC-25 Wallet Interation Standard.
+
+## Terminology
+
+* wallet: A service that manages a user's keys and can sign and perform canister calls on their behalf.
+* relying party: A service that wants to request calls on a specific canister.
+
+## Messages
+
+### `permission`
+
+Additional `ledger` scope is passed in the request and in the response an additional `subaccounts` property is returned in each `identity` when the wallet supports this extension.
+
+There are many types of wallets:
+- Wallets that support many different standards
+- Wallets that support multiple networks
+- Wallets that keep a list of subaccounts per ledger
+- Wallets that keep a list of subaccounts used across all ledgers
+
+#### Request
+
+The `ledger` scope is passed to the ICRC-25 `permission` request message scopes, no additional fields in the request are required in this spec.
+
+#### Response
+
+`identities`: Same as ICRC-25 response with additional `ledger` field.
+  - `subaccounts` (`vec`): List of ledger subaccounts for this identity.
+    - `networks` (`vec text`): List of network chain ids for this subaccount, this should be equal to or a subset of the networks in the permission request.
+    - `standards` (`vec text`): List of standards for this subaccount
+    - `canisterIds` (`opt vec principal`): Optional list of ledger canister ids for this subaccount, subaccount is assumed to be for any supported ledger if undefined.
```

Comment:
Maybe `ledgers` is a better naming?

Reply:
I think canisterIds is actually more accurate.

Reply:
I agree, naming it `ledgers` would also be confusing with the `icrc1_ledger` parent field.